internal/filter/filter_test.go

```diff
@@ -226,6 +228,49 @@ func TestChildMatch(t *testing.T) {
 	}
 }
 
+func TestValidPatterns(t *testing.T) {
+	// Test invalid patterns are correctly detected and returned
+	t.Run("detect-invalid-patterns", func(t *testing.T) {
+		allValid, invalidPatterns := filter.ValidatePatterns([]string{"*.foo", "*[._]log[.-][0-9]"})
+
+		rtest.Assert(t, allValid == false,
+			"Expected invalid patterns to be detected")
+
+		rtest.Assert(t, reflect.DeepEqual(invalidPatterns, []string{"*[._]log[.-][0-9]"}),
```

Comment:
`rtest.Equals` would make debugging the test easier

Reply:
Changed it accordingly 👍🏻